app/services/katello/pulp3/repository/yum.rb

```diff
@@ -12,11 +12,12 @@ class Yum < ::Katello::Pulp3::Repository
 
         def remote_options
           options = common_remote_options
-          uri = URI(root.url)
+          uri = root.url ? URI(root.url) : nil
           unless root.upstream_authentication_token.blank?
             options.merge!(sles_auth_token: root.upstream_authentication_token)
           end
-          options.merge!(url: uri.to_s, policy: root.download_policy)
+          options.merge!(url: uri&.to_s) if uri
```

Comment:
Minor. `&` seems not necessary after checking the existence of `uri`

Reply:
Ya..makes sense..playing it too safe..:smile: Will update..